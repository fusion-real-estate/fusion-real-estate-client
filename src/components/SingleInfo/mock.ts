export default {
  title: 'Luxury Family Loft by Victoria Park',
  subtitle: 'Quincy St, Brooklyn, NY, USA',
  feature: 'Destaque',
  status: 'Venda',
  offer: 'Oferta',
  beds: '4 Quartos',
  bath: '1 Banheiros',
  garage: '1 Garagem',
  sqt: '150 Metros',
  price: '1200.000,00',
  type: 'Apartamento'
}
