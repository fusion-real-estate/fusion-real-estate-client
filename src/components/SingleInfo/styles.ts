import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    padding: ${theme.spacings.xsmall};
    background-color: ${theme.colors.light};
    display: flex;
    justify-content: space-between;
  `}
`

export const Title = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.normal};
    padding-right: ${theme.spacings.xsmall};
  `}
`

export const BoxFeatures = styled.div`
  display: flex;
  align-items: center;
`

export const StylesBox = styled.div`
  display: flex;
  align-items: center;

  ${({ theme }) => css`
    > div {
      width: 8rem;
      text-align: center;
      padding: 0.3rem 0.5rem;
      color: ${theme.colors.white};
      font-size: ${theme.font.sizes.xsmall};
      border-radius: ${theme.border.radius};
      margin-right: ${theme.spacings.xxsmall};
    }
  `}
`

export const Feature = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondary};
  `}
`

export const Status = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
  `}
`

export const Offer = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.hot};
  `}
`

export const BoxInformation = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    margin-top: ${theme.spacings.xsmall};
  `}
`

export const BoxList = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    margin-right: ${theme.spacings.xxsmall};

    svg {
      height: 2rem;
      width: 2rem;
      color: ${theme.colors.gray};
      margin-right: ${theme.spacings.xsmall};
    }
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-weight: ${theme.font.normal};
    font-size: ${theme.font.sizes.xsmall};
  `}
`

export const SingleContainer = styled.div`
  text-align: left;
`

export const BuyTypeBox = styled.div`
  text-align: right;
`

export const Price = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: ${theme.font.bold};
  `}
`

export const Type = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.normal};
  `}
`
