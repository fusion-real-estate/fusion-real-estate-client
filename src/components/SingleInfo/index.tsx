import Heading from 'components/Heading'

import Icon from '@mdi/react'

import { mdiBedKingOutline, mdiShower, mdiCar, mdiFormTextarea } from '@mdi/js'

import formatPrice from 'utils/format-price'

import * as S from './styles'

export type SingleInfoProps = {
  title: string
  subtitle: string
  status?: string
  offer?: string
  beds: number
  bath: number
  garage: number
  sqt: number
  price: number
  type: string
}

const SingleInfo = ({
  title,
  subtitle,
  status,
  offer,
  beds,
  bath,
  garage,
  sqt,
  price,
  type
}: SingleInfoProps) => (
  <S.Wrapper>
    <S.SingleContainer>
      <Heading>{title}</Heading>

      <S.BoxFeatures>
        <S.Title>{subtitle}</S.Title>

        <S.StylesBox>
          <S.Status>{status}</S.Status>
          <S.Offer>{offer}</S.Offer>
        </S.StylesBox>
      </S.BoxFeatures>

      <S.BoxInformation>
        <S.BoxList>
          <Icon path={mdiBedKingOutline} />
          <S.Description>
            {beds === 1 ? `${beds} Quarto` : `${beds} Quartos`}
          </S.Description>
        </S.BoxList>
        <S.BoxList>
          <Icon path={mdiShower} />
          <S.Description>
            {bath === 1 ? `${bath} Banheiro` : `${bath} Banheiros`}
          </S.Description>
        </S.BoxList>
        <S.BoxList>
          <Icon path={mdiCar} />
          <S.Description>
            {garage === 1 ? `${garage} Garagem` : `${garage} Garagens`}
          </S.Description>
        </S.BoxList>
        <S.BoxList>
          <Icon path={mdiFormTextarea} />
          <S.Description>
            {sqt === 1 ? `${sqt} Metro(m²)` : `${sqt} Metros(m²)`}
          </S.Description>
        </S.BoxList>
      </S.BoxInformation>
    </S.SingleContainer>

    <S.BuyTypeBox>
      <S.Price>{formatPrice(price)}</S.Price>
      <S.Type>{type}</S.Type>
    </S.BuyTypeBox>
  </S.Wrapper>
)

export default SingleInfo
