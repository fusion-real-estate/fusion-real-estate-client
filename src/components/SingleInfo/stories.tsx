import { Story, Meta } from '@storybook/react/types-6-0'
import SingleInfo, { SingleInfoProps } from '.'

import mockSingleInfo from './mock'

export default {
  title: 'Page Single/SingleInfo',
  component: SingleInfo,
  args: mockSingleInfo
} as Meta

export const Basic: Story<SingleInfoProps> = (args) => (
  <div style={{ maxWidth: '144rem', padding: '1.5rem', margin: '0 auto' }}>
    <SingleInfo {...args} />
  </div>
)
