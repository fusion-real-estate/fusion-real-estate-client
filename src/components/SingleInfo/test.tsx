import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import SingleInfo from '.'

const props = {
  title: 'Title',
  subtitle: 'Subtitle',
  feature: 'Destaque',
  status: 'Venda',
  offer: 'Oferta',
  beds: '4',
  bath: '1',
  garage: '1',
  sqt: '150',
  price: 1200.0,
  type: 'Apartamento'
}

describe('<SingleInfo />', () => {
  it('should render single information', () => {
    const { container } = renderWithTheme(<SingleInfo {...props} />)

    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument()

    expect(screen.getByText(/subtitle/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render single information features', () => {
    renderWithTheme(<SingleInfo {...props} />)

    expect(screen.getByText(/destaque/i)).toBeInTheDocument()
    expect(screen.getByText(/venda/i)).toBeInTheDocument()
    expect(screen.getByText(/oferta/i)).toBeInTheDocument()
  })
})
