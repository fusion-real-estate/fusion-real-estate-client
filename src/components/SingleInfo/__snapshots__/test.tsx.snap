// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleInfo /> should render single information 1`] = `
.c2 {
  color: #373737;
  font-size: 2.0rem;
}

.c0 {
  padding: 1.6rem;
  background-color: #F3F3F3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  color: #7D7D7D;
  font-size: 1.8rem;
  font-weight: 400;
  padding-right: 1.6rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > div {
  width: 8rem;
  text-align: center;
  padding: 0.3rem 0.5rem;
  color: #FAFAFA;
  font-size: 1.2rem;
  border-radius: 0.4rem;
  margin-right: 0.8rem;
}

.c6 {
  background-color: #34ABD0;
}

.c7 {
  background-color: #0D263B;
}

.c8 {
  background-color: #CC2828;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1.6rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 0.8rem;
}

.c10 svg {
  height: 2rem;
  width: 2rem;
  color: #7D7D7D;
  margin-right: 1.6rem;
}

.c11 {
  color: #7D7D7D;
  font-weight: 400;
  font-size: 1.2rem;
}

.c1 {
  text-align: left;
}

.c12 {
  text-align: right;
}

.c13 {
  color: #0D263B;
  font-size: 2.8rem;
  font-weight: 600;
}

.c14 {
  color: #7D7D7D;
  font-size: 1.8rem;
  font-weight: 400;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="black"
    >
      Title
    </h2>
    <div
      class="c3"
    >
      <p
        class="c4"
      >
        Subtitle
      </p>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          Destaque
        </div>
        <div
          class="c7"
        >
          Venda
        </div>
        <div
          class="c8"
        >
          Oferta
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class="c10"
      >
        <svg
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 10V7A2 2 0 0 0 18 5H6A2 2 0 0 0 4 7V10A2 2 0 0 0 2 12V17H3.33L4 19H5L5.67 17H18.33L19 19H20L20.67 17H22V12A2 2 0 0 0 20 10M13 7H18V10H13M6 7H11V10H6M20 15H4V12H20Z"
            style="fill: currentColor;"
          />
        </svg>
        <p
          class="c11"
        >
          4
        </p>
      </div>
      <div
        class="c10"
      >
        <svg
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M21,14V15C21,16.91 19.93,18.57 18.35,19.41L19,22H17L16.5,20C16.33,20 16.17,20 16,20H8C7.83,20 7.67,20 7.5,20L7,22H5L5.65,19.41C4.07,18.57 3,16.91 3,15V14H2V12H20V5A1,1 0 0,0 19,4C18.5,4 18.12,4.34 18,4.79C18.63,5.33 19,6.13 19,7H13A3,3 0 0,1 16,4C16.06,4 16.11,4 16.17,4C16.58,2.84 17.69,2 19,2A3,3 0 0,1 22,5V14H21V14M19,14H5V15A3,3 0 0,0 8,18H16A3,3 0 0,0 19,15V14Z"
            style="fill: currentColor;"
          />
        </svg>
        <p
          class="c11"
        >
          1
        </p>
      </div>
      <div
        class="c10"
      >
        <svg
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M5,11L6.5,6.5H17.5L19,11M17.5,16A1.5,1.5 0 0,1 16,14.5A1.5,1.5 0 0,1 17.5,13A1.5,1.5 0 0,1 19,14.5A1.5,1.5 0 0,1 17.5,16M6.5,16A1.5,1.5 0 0,1 5,14.5A1.5,1.5 0 0,1 6.5,13A1.5,1.5 0 0,1 8,14.5A1.5,1.5 0 0,1 6.5,16M18.92,6C18.72,5.42 18.16,5 17.5,5H6.5C5.84,5 5.28,5.42 5.08,6L3,12V20A1,1 0 0,0 4,21H5A1,1 0 0,0 6,20V19H18V20A1,1 0 0,0 19,21H20A1,1 0 0,0 21,20V12L18.92,6Z"
            style="fill: currentColor;"
          />
        </svg>
        <p
          class="c11"
        >
          1
        </p>
      </div>
      <div
        class="c10"
      >
        <svg
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 9H4V23H2V7H13M23 9V7H17V5A1 1 0 0 1 18 4H20V2H17.5C17 2 16 2.45 16 3C16 2.45 15.05 2 14.5 2H12V4H14A1 1 0 0 1 15 5V19A1 1 0 0 1 14 20H12V22H14.5C15.05 22 16 21.55 16 21C16 21.55 16.95 22 17.5 22H20V20H18A1 1 0 0 1 17 19V9Z"
            style="fill: currentColor;"
          />
        </svg>
        <p
          class="c11"
        >
          150
        </p>
      </div>
    </div>
  </div>
  <div
    class="c12"
  >
    <div
      class="c13"
    >
      R$Â 1.200,00
    </div>
    <div
      class="c14"
    >
      Apartamento
    </div>
  </div>
</main>
`;
